src/tools/illink/src/ILLink.Shared/TrimAnalysis/HandleCallAction.cs

```diff
@@ -1216,6 +1278,23 @@ private IEnumerable<MultiValue> ProcessGetMethodByName (TypeProxy type, string m
 				yield return NullValue.Instance;
 		}
 
+		private IEnumerable<MultiValue> ProcessGetMethodByNameWithPrefix (TypeProxy type, string methodName, bool nameIsPrefix, BindingFlags? bindingFlags)
```

Comment:
Do we think we need to recognize the prefix lookups intrinsically? It seems to bring complexity, but the benefit is questionable (How common is it, to use the prefix lookup functionality - I've never heard about it before this PR honestly). I think it would make sense to treat that case as "unsupported" and revert to the "keep all behavior" in that case.

There will probably always be cases which the trimming doesn't fully implement (it just doesn't make sense to invest resources into emulating every little detail of the runtime behavior, even if possible). As such it makes sense to consciously only implement the things which bring enough value. There's cost associated even with good code, we have to maintain it going forward, future us (and others) will re-read this code to understand and so on.

Same logic applies to the various `MemberType` values.

To put it simply - I don't believe there's value in implementing all of the functionality the runtime has. This is not about correctness, it's just about size. Intentionally leaving certain parts as "unrecognized" will not break applications, just make them larger.

Reply:
@vitek-karas I use the `MemberType` values, and it was why I made the issue in the first place...

I can remove the prefix lookup code if we believe it overcomplicates things, as I don't think it provides a lot of value.

Reply:
@hamarb123 makes sense. I wasn't commenting on the whole change - specifically I was unsure about the prefixes. Removing that would make sense to me - and it should simplify the change (especially the shred code changes).